api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -310,6 +310,13 @@ message HttpConnectionManager {
   // response is initiated. If not specified or set to 0, this timeout is disabled.
   google.protobuf.Duration request_timeout = 28;
 
+  // The maximum duration of a connection. The duration is defined as a period since a connection
+  // was established. If not set, there is no max duration. When max_connection_duration is reached
+  // the connection will be closed. If the connection is an HTTP/2 connection a drain sequence will
+  // occur prior to closing the connection. See :ref:`drain_timeout
+  // <envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.drain_timeout>`.
+  google.protobuf.Duration max_connection_duration = 36;
```

Comment:
If we might want this one day on the upstream, I would prefer we move it to the protocol options. If we are certain it is only ever a downstream thing (which I would find surprising), then this is fine.

Reply:
We might want to upper-bound upstream connections as well.
What would be the best way to mark is as 'not implemented' for upstream connections?

Reply:
My preference is if we're implementing it in one place to have it here, and if we move it that we implement in both places (which is a fairly big ask)
Happy to have @mattklein123 tie break if we prefer having it half-implemented in the right place, vs in the potentially wrong place but have config clarity :-)

Reply:
My vote would be to move it into the common options if we think that it will eventually be implemented for upstream, mainly because the deprecation cycle is pretty painful.

Reply:
I do share @alyssawilk 's concern over features being added to the API with no intention of Envoy-side support, but that's mostly focused on the situation where an API is featured with zero Envoy implementation. Having a partial implementation that delivers value, together with a TODO and tracking issue so that someone who cares about the behavior can follow-thru on later seems a lot more tenable to me.

Reply:
I defined max_connection_duration in common options and added some stubs in upstream implementation.